### PR TITLE
Replace a.i-button with button

### PR DESCRIPTION
**Pull request by bxo11**
PR contains updated instances of a.i-button that can be replaced with a button tag.

### CODE REVIEW

File: indico/modules/attachments/templates/_attachments.html [hunk 1]
```diff
@@ -143,9 +143,9 @@
     {%- endif -%}
     {% if management %}
         <div class="i-box-footer text-right js-attachment-actions hide-if-locked">
-            <a href="#" class="i-button icon-folder-plus js-dialog-action"
+            <button class="i-button icon-folder-plus js-dialog-action"
                data-href="{{ url_for('attachments.create_folder', linked_object) }}"
-               data-title="{% trans %}Create a new folder{% endtrans %}">{% trans %}New folder{% endtrans -%}</a>
+               data-title="{% trans %}Create a new folder{% endtrans %}">{% trans %}New folder{% endtrans -%}</button>
```
Comment (ThiefMaster) [inline indico/modules/attachments/templates/_attachments.html:148]: Can you please fix the indentation here (and in other places if necessary)? We typically align the attributes below the first attribute, ie like this:

```html
<button class="..."
        foo="bar">
```

**Comment by ThiefMaster** [comment]
Thanks! Did you already test all the places where you changed it to make sure nothing breaks?

File: indico/modules/events/editing/client/js/management/editable_type/EditableTypeDashboard.jsx [hunk 2]
```diff
@@ -93,9 +93,9 @@ export default function EditableTypeDashboard() {
               }
               description={Translate.string('Toggle whether users can submit new editables')}
             >
-              <a className="i-button highlight" onClick={toggleSubmission}>
+              <button className="i-button highlight" onClick={toggleSubmission} type="button">
```
Comment (bxo11) [inline indico/modules/events/editing/client/js/management/editable_type/EditableTypeDashboard.jsx:96]: I am receiving here a warning that buttons in `EditableTypeDashboard.jsx` file need a type. Is the `button` type right for these elements?

**Comment by ThiefMaster** [comment]
Hey, sorry, but your last commit also indents the children of buttons. After rebasing your PR to the latest master, I started fixing some of them but then realized how many places you changed, so I'll have to ask you to clean up the rest of them...

**Comment by bxo11** [comment]
I cleaned up the changes and indents

**Review by ThiefMaster — APPROVED** [review]
I made some fixes hat caused weird display issues while testing, usually due to introducing whitespace in previously-empty elements or looked weird (mixing buttons and a.i-buttons next to each other does not always look the same)

**Pull request merged**